### PR TITLE
feat(alerts): Emit `alert.sent` analytic on incidents

### LINKED ISSUE #54417: Add notification_uuid to IncidentActivity
Add a notification_uuid column to IncidentActivity.
Passed as notification_uuid as a query parameter in URLs

AC:
- Each time we make an entry of IncidentActivity, we should generate and save an `notification_uuid` which is the UUID for the notification sent. Since each IncidentActivity maps directly to sending a notification for resolved/critical/warning.
- Passed as `notification_uuid` as query parameter in URLs

### PR DESCRIPTION
**Pull request by scttcper**
fixes #54417

### CODE REVIEW

File: src/sentry/analytics/events/alert_sent.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from sentry import analytics
+
+
+class AlertSentEvent(analytics.Event):
+    type = "alert.sent"
+
+    attributes = (
+        analytics.Attribute("organization_id"),
+        analytics.Attribute("project_id", required=False),
```
Comment (scefali) [inline src/sentry/analytics/events/alert_sent.py:9]: @scttcper when would we not have a project_id?

Comment (scttcper) [reply src/sentry/analytics/events/alert_sent.py:9]: i'm not sure, the other notification_sent analytic had it and i started with that

Comment (scefali) [reply src/sentry/analytics/events/alert_sent.py:9]: @scttcper can we make it required? I think that's better.

File: src/sentry/incidents/action_handlers.py [hunk 2]
```diff
@@ -57,6 +58,20 @@ def resolve(
     ):
         pass
 
+    def record_sent_analytics(
```
Comment (jangjodi) [inline src/sentry/incidents/action_handlers.py:61]: nit: can we call this `record_alert_sent_analytics`?

File: src/sentry/integrations/pagerduty/utils.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ def send_incident_alert_notification(
     metric_value: int,
     new_status: IncidentStatus,
     notification_uuid: str | None = None,
-) -> None:
+) -> bool:
```
Comment (jangjodi) [inline src/sentry/integrations/pagerduty/utils.py:65]: does this not cause a typing error since a `bool` isn't always returned? same with `/opgenie/`

Comment (scttcper) [reply src/sentry/integrations/pagerduty/utils.py:65]: for some reason some raise and others don't

**Review by jangjodi — APPROVED** [review]

**Pull request merged**